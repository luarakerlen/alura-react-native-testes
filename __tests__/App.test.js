import React from 'react';
import { render } from '@testing-library/react-native';
import { View } from 'react-native';
import AppRotas from '../src/rotas/AppRotas';
import App from '../App';

jest.mock('../src/rotas/AppRotas');

describe('App', () => {
	beforeAll(() => {
		AppRotas.mockImplementation(() => <View testID='app-rotas-id' />);
	});

	it('deve renderizar corretamente', () => {
		const { toJSON } = render(<App />);
		expect(toJSON()).toMatchSnapshot();
	});
});
