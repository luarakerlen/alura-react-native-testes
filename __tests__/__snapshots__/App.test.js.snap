// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App deve renderizar corretamente 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      testID="app-rotas-id"
    />
  </View>
</RCTSafeAreaView>
`;
